### PR TITLE
PIM-7901: Fix memory leak on "compute_family_variant_structure_changes" job

### PR DESCRIPTION
**Pull request by Doodoune**
**Description (for Contributor and Core Developer)**

The "ComputeFamilyVariantStructureChangesTasklet" try to make a saveAll on the product repository with too many products in memory (for our client is 6500)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php [hunk 1]
```diff
@@ -125,11 +138,39 @@ private function updateValuesOfEntities(array $entities): void
...
+    {
+        if (count($productModels) > $this->batchSize)
+        {
+            while (count($productModels) > 0) {
+                $this->productModelSaver->saveAll(array_splice($productModels, 0, $this->batchSize));
+                $this->cacheClearer->clear();
+            }
+        }
+        else {
+            $this->productModelSaver->saveAll($productModels);
```
Comment (SamirBoulil) [inline src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php]: don't you cacheClear here ?

File: src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php [hunk 2]
```diff
@@ -125,11 +138,39 @@ private function updateValuesOfEntities(array $entities): void
...
+    {
+        if (count($products) > $this->batchSize)
+        {
+            while (count($products) > 0) {
+                $this->productSaver->saveAll(array_splice($products, 0, $this->batchSize));
+                $this->cacheClearer->clear();
+            }
+        }
+        else {
             $this->productSaver->saveAll($products);
```
Comment (SamirBoulil) [inline src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php]: same ?

File: src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php (hunk 1 above)
Comment (rouliane) [reply src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php]: :+1:  I agree, we should also clear the cache here

Comment (Doodoune) [reply src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php]: After a problem detected with cache clearer, I decided to remove the CacheClearer.
It works on the client env.

File: src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php (hunk 2 above)
Comment (Doodoune) [reply src/Pim/Component/Catalog/Job/ComputeFamilyVariantStructureChangesTasklet.php]: After a problem detected with cache clearer, I decided to remove the CacheClearer.
It works on the client env.

**Pull request merged**